### DISCUSSION TITLE
Leaf compilation fails

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#528] [Q&A] [answered]
Hi Folks

I'm following the [4 - Compile the Leaf API instructions](https://leafdocs.rit.uw.edu/installation/installation_steps/4_compile_api/).

Environment:

![envt](https://user-images.githubusercontent.com/33882/171894618-ca069d69-0dad-4b18-b715-b73ff00d707c.png)

`dotnet publish -c Release -o /var/opt/leafapi/api` generates this output (attaching screenshot to retain colors):

![compile_fails](https://user-images.githubusercontent.com/33882/171897982-a53f504a-5d91-4db3-93af-4b83c2ee16e0.png)

Your diagnosis, please?

Thanks
Arthur

**Comment by ndobb** [comment, marked as answer]
Grasping for straws - what version of `dotnet` is installed, Arthur?

Here's what I'm using:
![image](https://user-images.githubusercontent.com/10424962/171954706-0e931f30-fef4-439b-a757-cb543416d0dc.png)

...which does not produce the same errors:
![image](https://user-images.githubusercontent.com/10424962/171954747-b3018c7a-dc2c-459c-917d-efe955e999ba.png)

Last idea: can you check to see if the offending lines mentioned in [Services/Compiler/LeafDbDataReader.cs](https://github.com/uwrit/leaf/blob/master/src/server/Services/Compiler/LeafDbDataReader.cs) are indeed the same as in [GitHub master](https://github.com/uwrit/leaf/blob/master/src/server/Services/Compiler/LeafDbDataReader.cs)?

**Reply by artgoldberg** [reply]
Hi Nic. Yeah, I also wondered about the `dotnet` version, but only briefly. I have:

    $ dotnet --version
    3.1.408

Now updated to

    $ dotnet --version
    6.0.300

And the previous error is gone. Great straw to grasp!
However, the results are mixed. 

`dotnet publish -c Release -o /var/opt/leafapi/api` works, which is excellent!

But `dotnet publish -c Debug -o /var/opt/leafapi/api` generates this error:
![debug_error](https://user-images.githubusercontent.com/33882/172199456-88b4bfde-de85-4718-9cdd-dc49ed90b107.png)

Text version:

    /usr/share/dotnet/sdk/6.0.300/Microsoft.Common.CurrentVersion.targets(5097,5): error MSB3030: Could not copy the file "/var/opt/leafapi/leaf_download/leaf/src/server/API/obj/Release/netcoreapp3.1/staticwebassets/API.StaticWebAssets.xml" because it was not found. [/var/opt/leafapi/leaf_download/leaf/src/server/Tests/Tests.csproj]

`netcoreapp3.1` contains this file about static assets:
![static_assets](https://user-images.githubusercontent.com/33882/172200524-19948dbc-07aa-419e-ab19-1e57db9c5924.png)

`dotnet publish -c Debug -o /var/opt/leafapi/api` also generates 294 warnings, but none of them seem relevant. 
[debug_warnings.txt](https://github.com/uwrit/leaf/files/8845813/debug_warnings.txt)



Thanks
Arthur

**Reply by artgoldberg** [reply]
Curiously, on another attempt `dotnet publish -c Debug -o /var/opt/leafapi/api` works:

![dotnet debug works](https://user-images.githubusercontent.com/33882/172926940-5424f484-f23a-4b13-91f4-1bc922d6faee.png)

**Reply by ndobb** [reply]
Got it, thanks Arthur. I'm glad to hear that worked the 2nd time but am not sure why it did not the 1st. I generally don't publish in `Debug` mode so that may be something we need to investigate.
